### ISSUE TITLE
xBridge and Wifi Wixel

### ISSUE THREAD

**Issue by danpowell88** [StephenBlackWasAlreadyTaken/xDrip-Experimental#341]
Is it possible to have an option for xbridge and wifi wixel, currently it looks like you need to use the original bt if you want a wifi wixel mesh

**Comment by tzachi-dar** [comment]
You are correct, it is not possible right now.
It seems that the needed change might not be big from code perspective but
it will require some testing.

In other words, there are two things that need to be work out:
1) Change xBridge to write the correct format (the format that dexterity is
using).
2) Make sure that power options are handled correctly. If your setup does
not sleep (always connected to power) this hopefully should not be a
problem.

Than, you will have to burn your transmitter id into the xDrip (can be done
through putty, or so).

After that, you will need to test that... (this might be the harder part).

Do you want me to create such a fw?

Thanks
Tzachi

**Comment by danpowell88** [comment]
Hmm I think I might be confusing things a bit.

I was just talking about the menu option that lets you select "Wifi Wixel + BT Wixel" and provide a list of receivers.

The setup that I'd like to have is 
-  Mobile xbridge powered by battery
- Home wixel mesh (using xdrip-wixel and some modified code)

I have the wixel mesh working such that they just need to be hit on the ip:50005.

If I select Wifi wixel only in the xdrip mobile settings I can see xdrip is getting the values correct so I know everything is all working correctly.

Ultimately I'd like to be able to run the xbridge when I'm mobile and also have the wifi mesh when I'm at home, currently it looks like my mobile setup would have to use the standard BT wixel code.

Perhaps the "List of receivers" field should always be available and the wifi connection methods be removed as an option and if anything is defined in list of receivers it should just hit that?

**Comment by danpowell88** [comment]
Its not idea but I've managed to hack the code so that the "Xbridge" setting behaves like the "BT + Wifi Wixel" setting but using an xbridge instead of the standard code.

Open to ideas on how to clean this up

**Comment by tzachi-dar** [comment]
There have been rumors that xBridge also works when xDrip is in xDrip mode. You just need to have it once in xBridge mode so it will ask for transmiter id and then it will start to work.
Did you check if it is working?

Thanks
Tzachi

**Comment by thecubical** [comment]
I'd like to use the xbridge in xbridge mode to get some of the nice features like autoretry on bluetooth failure which I don't think the default wixel code supports

**Comment by jstevensog** [comment]
There is no auto retry in the xBridge connection code.  It is identical and
part of the same module as the dexdrip bridge code.
Cheers

On Mon, May 30, 2016 at 12:46 PM, Daniel Powell notifications@github.com
wrote:

> I'd like to use the xbridge in xbridge mode to get some of the nice
> features like autoretry on bluetooth failure which I don't think the
> default wixel code supports
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/issues/341#issuecomment-222401548,
> or mute the thread
> https://github.com/notifications/unsubscribe/AIQs81VLTHNdDcoLiUZSgoGmuTr31m9Aks5qGk-SgaJpZM4ImfW-
> .

## 

John Stevens
"You are how you live, not what you have."

**Comment by danpowell88** [comment]
So the main difference is just that you can configure the transmitter id at runtime?

**Comment by jstevensog** [comment]
No. That is a big plus, obviously. But xBridge2 uses a binary protocol rather than plain text, he's a better battery life, and includes battery monitoring.
Cheers

---- Daniel Powell wrote ----

**Comment by tzachi-dar** [comment]
It looks like adding one more option to the code of wifi + xBridge should not be complicated.
If I'll send you such an apk, will you test that Daniel?

**Comment by danpowell88** [comment]
Yeah no problem

**Comment by tzachi-dar** [comment]
Here is the debug apk.
Please rename it to .apk and install.

Let me know if it is OK or there are problems.
![app-xdrip-debug](https://cloud.githubusercontent.com/assets/7942372/15682364/f5069158-2765-11e6-88eb-7c713c50a2d2.png)

**Comment by tzachi-dar** [comment]
Please hold your testing, I think I have found an issue that I want to understand first.

**Comment by tzachi-dar** [comment]
![app-xdrip-debug](https://cloud.githubusercontent.com/assets/7942372/15689050/c8fc00aa-2785-11e6-96ba-8b9b7b92fed0.jpg)
Here is the updated file.
Please let us know how it is working.

**Comment by danpowell88** [comment]
It wont let me install over the top of Beta5 v2.0.5_2 update 2, will probably have to wait until the weekend when I can uninstall and reinstall from fresh.

Cheers

**Comment by tzachi-dar** [comment]
![app-xdrip-release](https://cloud.githubusercontent.com/assets/7942372/15715761/5c254a92-2828-11e6-9e6d-acf83f658bf3.jpg)
Here is the signed release binary. I believe that it will let you install it over the beta.

**Comment by danpowell88** [comment]
Got it working, will test over the next few days

**Comment by ondrejdiabetes** [comment]
could you send me the apk file as well?

**Comment by AdrianLxM** [comment]
@ondrejdiabetes "safe link as" still works. But this has already made the main app month ago, I believe?
